frontend/src/screens/wallet/Send.tsx

```diff
@@ -16,98 +10,123 @@ import {
-  const [paymentDone, setPaymentDone] = React.useState(false);
 
-  if (!balances || !channels) {
-    return <Loading />;
-  }
+  const [amount, setAmount] = React.useState<number>(0);
+  const [recipient, setRecipient] = React.useState<string>("");
+  const [lnurlDetails, setLnurlDetails] = React.useState<
+    Invoice | LightningAddress
+  >();
 
-  const handleContinue = () => {
-    try {
-      setInvoiceDetails(new Invoice({ pr: invoice }));
-    } catch (error) {
-      toast({
-        variant: "destructive",
-        title: "Invalid Payment Request",
-      });
-      console.error(error);
-      setInvoice("");
-    }
+  const paste = async () => {
+    const text = await navigator.clipboard.readText();
+    setRecipient(text.trim());
   };
 
-  const handleSubmit = async (event: React.FormEvent<HTMLFormElement>) => {
+  const onSubmit = async (event: React.FormEvent<HTMLFormElement>) => {
     event.preventDefault();
     try {
-      setLoading(true);
-      const payInvoiceResponse = await request<PayInvoiceResponse>(
-        `/api/payments/${invoice}`,
-        {
-          method: "POST",
-          headers: {
-            "Content-Type": "application/json",
-          },
+      if (LIGHTNING_ADDRESS_REGEX.test(recipient)) {
+        const lnAddress = new LightningAddress(recipient);
+        await lnAddress.fetch();
+        if (!lnAddress.lnurlpData) {
+          throw new Error("invalid lightning address");
         }
-      );
-      if (payInvoiceResponse) {
-        setPayResponse(payInvoiceResponse);
-        setPaymentDone(true);
-        setInvoice("");
-        toast({
-          title: "Successfully paid invoice",
-        });
+        setLnurlDetails(lnAddress);
+      } else if (BOLT11_REGEX.test(recipient)) {
+        const invoice = new Invoice({ pr: recipient });
+        setAmount(invoice.satoshi);
+        setLnurlDetails(invoice);
+      } else {
+        throw new Error("invalid recipient");
       }
-    } catch (e) {
+    } catch (error) {
       toast({
         variant: "destructive",
-        title: "Failed to send: " + e,
+        title: "Failed to send payment",
+        description: "" + error,
       });
-      setInvoice("");
-      setInvoiceDetails(null);
-      console.error(e);
+      console.error(error);
     }
-    setLoading(false);
   };
 
-  const copy = () => {
-    copyToClipboard(payResponse?.preimage as string);
-    toast({ title: "Copied to clipboard." });
+  const onReset = async () => {
+    setAmount(0);
+    setRecipient("");
+    setLnurlDetails(undefined);
   };
 
-  const paste = async () => {
-    const text = await navigator.clipboard.readText();
-    setInvoice(text.trim());
+  if (!balances || !channels) {
+    return <Loading />;
+  }
+
+  const hasLowBalance =
+    hasChannelManagement &&
+    (amount || 0) * 1000 >= 0.8 * balances.lightning.totalSpendable;
+
+  const renderScreen = () => {
+    if (lnurlDetails instanceof LightningAddress) {
```

Comment:
the flow seems kind of weird here, that the payment success cards are rendered in the individual components. A success is a success, it doesn't change based on which payment type, right?